ngraph/core/reference/include/ngraph/runtime/reference/max.hpp

```diff
@@ -17,35 +18,33 @@ namespace ngraph
         namespace reference
         {
             template <typename T>
-            void max(const T* arg,
-                     T* out,
-                     const Shape& in_shape,
-                     const AxisSet& reduction_axes,
-                     bool keep_dims)
+            void max(const T* arg, T* out, const Shape& in_shape, const AxisSet& reduction_axes)
             {
                 T minval = std::numeric_limits<T>::has_infinity
                                ? T(-std::numeric_limits<T>::infinity())
                                : std::numeric_limits<T>::min();
 
-                auto out_shape = reduce(in_shape, reduction_axes, keep_dims);
-                CoordinateTransform output_transform(out_shape);
+                auto out_shape = reduce(in_shape, reduction_axes, false);
+                std::fill(out, out + shape_size(out_shape), minval);
 
-                for (const Coordinate& output_coord : output_transform)
-                {
-                    out[output_transform.index(output_coord)] = minval;
-                }
+                const auto in_strides = row_major_strides(in_shape);
+                const auto out_strides = row_major_strides(out_shape);
 
                 CoordinateTransform input_transform(in_shape);
```

Comment:
Can we avoid `CoordinateTransform`?

Reply:
We discussed this in different op, ReduceSum. The conclusion was that we cannot avoid it fully. Here the [comment](https://github.com/openvinotoolkit/openvino/pull/5728#discussion_r639599572)